cmake/mcuboot.cmake

```diff
@@ -72,15 +72,11 @@ if(CONFIG_BOOTLOADER_MCUBOOT)
       )
   endfunction()
 
-  if (CONFIG_MCUBOOT_BUILD_S1_VARIANT)
-    # Inject this configuration from parent image to mcuboot.
-    set(conf_path "${ZEPHYR_NRF_MODULE_DIR}/subsys/bootloader/image/build_s1.conf")
-    string(FIND ${mcuboot_OVERLAY_CONFIG} ${conf_path} out)
-    if (${out} EQUAL -1)
-      set(mcuboot_OVERLAY_CONFIG
-        "${mcuboot_OVERLAY_CONFIG} ${conf_path}"
-        CACHE STRING "" FORCE)
-    endif()
+  if (CONFIG_BUILD_S1_VARIANT
```

Comment:
This change allows us to not use the "build_s1.conf" overlay file hack to copy the value of this variable from the app to mcuboot